spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/web/SpringDataWebProperties.java

```diff
@@ -81,6 +83,19 @@ public static class Pageable {
 		 */
 		private int maxPageSize = 2000;
 
+		/**
+		 * Configures how to render {@link org.springframework.data.domain.PageImpl}
+		 * instances. Defaults to {@link PageSerializationMode#DIRECT} for backward
+		 * compatibility reasons. Prefer explicitly setting this to
+		 * {@link PageSerializationMode#VIA_DTO}, or manually convert
+		 * {@link org.springframework.data.domain.PageImpl} instances before handing them
+		 * out of a controller method, either by manually calling
+		 * {@code new PagedModel<>(page)} or using Spring HATEOAS
+		 * {@link org.springframework.hateoas.PagedModel} abstraction.
+		 * @since 3.3
+		 */
+		private PageSerializationMode serializationMode = PageSerializationMode.DIRECT;
```

Comment:
The annotation processor the generates configuration property metadata can't handle enums due to a limitation of the annotation processing model. Please add an entry to `additional-spring-configuration-metadata.json` for this property's default value.